lib/src/clixon_xml_map.c

```diff
@@ -1588,6 +1588,226 @@ yang_enum2int(yang_stmt *ytype,
     return retval;
 }
 
+/*! Given a YANG (bits) type node and a bit string, return the bit position.
+ * Example:
+ *
+ * type bits {
+ *     bit stateA {
+ *         position "0"; << This one
+ *     }
+ *     bit stateB {
+ *         position "1"; << And this one
+ *     }
+ * }
+ * 
+ * If the position is not specified, it will be automatically assigned as defined
+ * in RFC7950, section 9.7.4.2
+ * 
+ * @param[in]  ytype  YANG type noden
+ * @param[in]  bitstr bit (flag) string
+ * @param[out] bitpos position for the given bit (flag)
+ * @retval     1      OK, result in flagpos
+ * @retval     0      Invalid, not found
+ * @retval    -1      Error
+ */
+int
+yang_bits_pos(yang_stmt *ytype,
+              char      *bitstr,
+              uint32_t  *bitpos)
+{
+    int retval = 0;
+    int ret = 0;
+    int is_first = 1;
+    char *reason;
+
+    yang_stmt *yprev = NULL;
+    yang_stmt *ypos = NULL;
+    while ((yprev = yn_each(ytype, yprev)) != NULL){
+        // Check for the given bit name (flag)
+        if (yang_keyword_get(yprev) == Y_BIT){
+            
+            // Use position from Y_POSITION statement if defined
+            if ((ypos = yang_find(yprev, Y_POSITION, NULL)) != NULL){
+                if ((ret = parse_uint32(yang_argument_get(ypos), bitpos, &reason)) < 0){
+                    clixon_err(OE_UNIX, EINVAL, "cannot parse bit position val: %s", reason);
+                    goto done;
+                }
+                if (ret == 0)
+                    goto fail;
+            } else {
+                // Position not defined. Use last known position + 1 (skip first node to start with 0)
+                if (is_first == 0) (*bitpos)++;
+            }
+
```

Comment:
Avoid extra empty lines in body code except after variable declarations.
This applies to many other instances elsewhere in this PR.